huaweicloud/services/aom/resource_huaweicloud_aom_alarm_rule.go

```diff
@@ -0,0 +1,418 @@
+		d.Set("threshold", rule.Threshold),
+		d.Set("unit", rule.Unit),
+	)
+
+	var dimensions []map[string]interface{}
+	for _, pairObject := range *rule.Dimensions {
+		dimension := make(map[string]interface{})
+		dimension["name"] = pairObject.Name
+		dimension["value"] = pairObject.Value
+
+		dimensions = append(dimensions, dimension)
+	}
+	mErr = multierror.Append(mErr, d.Set("dimensions", dimensions))
+
+	if err := mErr.ErrorOrNil(); err != nil {
+		return fmtp.DiagErrorf("error setting AOM alarm rule fields: %w", err)
+	}
+
+	return nil
+}
+
+func resourceAlarmRuleUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.HcAomV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("error creating AOM client: %s", err)
+	}
+
+	updateOpts := aom.UpdateAlarmRuleParam{
+		AlarmRuleName: d.Get("alarm_name").(string),
+	}
+
+	if d.HasChange("alarm_description") {
+		updateOpts.AlarmDescription = utils.String(d.Get("alarm_description").(string))
+	}
+	if d.HasChange("alarm_level") {
+		alarmLevel := d.Get("alarm_level").(int)
+		var alarmLevelToReq aom.UpdateAlarmRuleParamAlarmLevel
+		switch alarmLevel {
+		case 1:
+			alarmLevelToReq = aom.GetUpdateAlarmRuleParamAlarmLevelEnum().E_1
+		case 2:
+			alarmLevelToReq = aom.GetUpdateAlarmRuleParamAlarmLevelEnum().E_2
+
+		case 3:
+			alarmLevelToReq = aom.GetUpdateAlarmRuleParamAlarmLevelEnum().E_3
+
+		case 4:
+			alarmLevelToReq = aom.GetUpdateAlarmRuleParamAlarmLevelEnum().E_4
+		}
+		updateOpts.AlarmLevel = &alarmLevelToReq
+	}
+
+	if d.HasChange("alarm_enabled") {
+		updateOpts.IdTurnOn = utils.Bool(d.Get("alarm_enabled").(bool))
+	}
+	if d.HasChange("alarm_actions") {
+		updateOpts.AlarmActions = buildActionOpts(d.Get("alarm_actions").([]interface{}))
+	}
+	if d.HasChange("ok_actions") {
+		updateOpts.OkActions = buildActionOpts(d.Get("ok_actions").([]interface{}))
+	}
+	if d.HasChange("insufficient_data_actions") {
+		updateOpts.InsufficientDataActions = buildActionOpts(d.Get("insufficient_data_actions").([]interface{}))
+	}
+	if d.HasChange("metric_name") {
+		updateOpts.MetricName = utils.String(d.Get("metric_name").(string))
+	}
+	if d.HasChange("namespace") {
+		updateOpts.Namespace = utils.String(d.Get("namespace").(string))
+	}
+	if d.HasChange("dimensions") {
+		dimensions := buildDimensionsOpts(d.Get("dimensions").([]interface{}))
+		updateOpts.Dimensions = &dimensions
+	}
+	if d.HasChange("period") {
+		updateOpts.Period = utils.Int32(int32(d.Get("period").(int)))
+	}
+	if d.HasChange("unit") {
+		updateOpts.Unit = utils.String(d.Get("unit").(string))
+	}
+	if d.HasChange("comparison_operator") {
+		updateOpts.ComparisonOperator = utils.String(d.Get("comparison_operator").(string))
+	}
+	if d.HasChange("statistic") {
+		statistic := d.Get("statistic").(string)
+		var statisticToReq aom.UpdateAlarmRuleParamStatistic
+		switch statistic {
+		case "maximum":
```

Comment:
suggest to add a function to convert statistic to struct

Reply:
fixed